src/messages/MFSMap.h

```diff
@@ -0,0 +1,59 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software 
+ * Foundation.  See file COPYING.
+ * 
+ */
+
+
+#ifndef CEPH_MFSMAP_H
+#define CEPH_MFSMAP_H
+
+#include "msg/Message.h"
+#include "mds/FSMap.h"
+#include "include/ceph_features.h"
+
+class MFSMap : public Message {
+ public:
+  epoch_t epoch;
+  bufferlist encoded;
+
+  version_t get_epoch() const { return epoch; }
+  bufferlist& get_encoded() { return encoded; }
+
+  MFSMap() : 
+    Message(CEPH_MSG_FS_MAP), epoch(0) {}
+  MFSMap(const uuid_d &f, FSMap *fsmap) :
+    Message(CEPH_MSG_FS_MAP), epoch(fsmap->get_epoch())
+  {
+    fsmap->encode(encoded, -1);
```

Comment:
This was just an oversight when this was derived from MMDSMap

Passing through -1 here doesn't break anything (no old clients know about MFSMap, and FSMap::encode doesn't know about any feature bits), so let's fix it separately to avoid the need to re-test this PR. https://github.com/ceph/ceph/pull/7998